packages/devtools_app/lib/src/debugger/flutter/scripts.dart

```diff
@@ -5,86 +5,106 @@
 import 'package:flutter/material.dart';
 import 'package:vm_service/vm_service.dart';
 
+import '../../flutter/common_widgets.dart';
 import '../../flutter/theme.dart';
-import 'common.dart';
+import '../../utils.dart';
+
+// TODO(devoncarew): Show the filtered count in the scripts header.
 
 /// Picker that takes a [ScriptList] and allows selection of one of the scripts
 /// inside.
 class ScriptPicker extends StatefulWidget {
   const ScriptPicker({
     Key key,
     @required this.scripts,
-    @required this.onSelected,
     @required this.selected,
+    @required this.onSelected,
   }) : super(key: key);
 
-  final ScriptList scripts;
-  final void Function(ScriptRef scriptRef) onSelected;
+  final List<ScriptRef> scripts;
   final ScriptRef selected;
+  final void Function(ScriptRef scriptRef) onSelected;
 
   @override
   ScriptPickerState createState() => ScriptPickerState();
 }
 
 class ScriptPickerState extends State<ScriptPicker> {
-  List<ScriptRef> _filtered;
-  TextEditingController filterController = TextEditingController();
+  final _filterController = TextEditingController();
+  List<ScriptRef> _filteredScripts = [];
 
   @override
   void initState() {
     super.initState();
 
-    if (_isNotLoaded) initFilter();
+    _updateFiltered();
   }
 
   @override
   void didUpdateWidget(ScriptPicker oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    if (_isNotLoaded) {
-      initFilter();
-    } else if (oldWidget.scripts != widget.scripts) {
-      updateFilter(filterController.text);
-    }
+    updateFilter();
```

Comment:
nit: it would be idiomatic to 
only call updateFilter if
`widget.scripts != oldWidget.scripts`
assuming that object equality is a safe way to check whether the list of scripts changed.

Reply:
OK, it looks like this method means that we're re-using the state object from an older statefulwidget.

I don't think object identify will work for us here (wrt the list of scripts). For the future though, we'd need to check both the list of scripts, and the filter controller have the same value? Or, the two fields on the `widget` - the scripts and the `selected` script?